virttest/libvirt_xml/vm_xml.py

```diff
@@ -882,9 +882,9 @@ def set_vm_vcpus(vm_name, vcpus, current=None, sockets=None, cores=None,
                             nodexml_list.append(nodexml)
                         if numa_number is not None and numa_number > 0:
                             vmcpu_xml.xmltreefile.create_by_xpath('/numa')
-                            vmcpu_xml.numa_cell = vmcpu_xml.dicts_to_cells(nodexml_list)
+                            vmcpu_xml.numa_cell = nodexml_list
```

Comment:
I think I understand what happend.

Reply:
Could you also make a change to line 862 too?
`nodexml = {}`  to `nodexml = NumaCellXML()`
Or it could fail when trying to assign a 'dict' to 'numa_cell'

Reply:
@chloerh  I quite did not follow this change, 
` if vmxml.cpu.numa_cell:` this check in line 859 seems to be unnecessary or it has to be handled in try block to handle node> previous node in the xml vs node changed by numa_number i believe



```
-                            nodexml = vmcpu_xml.numa_cell[node]
+                            if vmxml.cpu.numa_cell:
+                                nodexml = vmcpu_xml.numa_cell[node]
+                            else:
+                                nodexml = {}
```

Reply:
Oh, it's actually about type of `nodexml` . It should be `NumaCellXML` type if it's going to be assigned to `numa_cell`. 
It is `dict` type by current code.

About the check of line 859, that's not added by me, I guess it was supposed to handle situations of 'no numa cell' ?

Reply:
@sathnaga as long as `nodexml` is of type `NumaCellXML`, I think it's ok